### PR TITLE
Add support for NIC datasource

### PR DESCRIPTION
**Pull request by taohe1012**
##### SUMMARY

Add support for nic datasource.
This Terraform datasource is used to query existing network interface cards(NIC) configuration including network adapters, network ports, network device functions and their OEM attributes.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
redfish network datasource

##### ADDITIONAL INFORMATION

```paste below

```

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

#### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```
image

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: gofish/dell/networkDeviceFunction.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+// This is the wrapper that extracts and parses Dell NetworkDeviceFunction OEM data,
+// as well as DellAssignablePhysicalPorts, DellAssignablePhysicalNetworkPorts, DellPhysicalPortAssignment and DellEthernet.
+func NetworkDeviceFunction(deviceFunction *redfish.NetworkDeviceFunction) (*NetworkDeviceFunctionExtended, error) {
+	dellNetworkDeviceFunction := &NetworkDeviceFunctionExtended{NetworkDeviceFunction: deviceFunction}
+
+	rawDataBytes, err := GetRawDataBytes(deviceFunction)
+	if err != nil {
+		return dellNetworkDeviceFunction, err
+	}
+	if oemRawData, found := GetNodeFromRawDataBytes(rawDataBytes, "Oem"); found == nil {
```
Comment (baoy1) [inline gofish/dell/networkDeviceFunction.go:303]: Many `GetNodeFromRawDataBytes`. Maybe can be refactored.

Comment (taohe1012) [reply gofish/dell/networkDeviceFunction.go:303]: thanks @baoy1 , will do that

Comment (rounak-adhikary) [reply gofish/dell/networkDeviceFunction.go:303]: `*redfish.NetworkDeviceFunction` type has a public field `Oem`. You can directly use that.

Comment (taohe1012) [reply gofish/dell/networkDeviceFunction.go:303]: hi @rounak-adhikary , for current gofish commit version, the NetworkDeviceFunction only has private rawData, no public  Oem. But in the latest gofish , *redfish.NetworkDeviceFunction type has a public field Oem.

**Review by rounak-adhikary — APPROVED** [review]

File: gofish/dell/networkDeviceFunction.go (hunk 1 above)
Comment (rounak-adhikary) [reply gofish/dell/networkDeviceFunction.go]: Okay, thanks!

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Pull request merged**